### ISSUE TITLE
invalid reassignment of non-absolute variable 'pos'

### ISSUE THREAD

**Issue by nickdesaulniers** [ClangBuiltLinux/linux#1043]
linux-next is full of errors for x86_64:
```
  AS      arch/x86/entry/entry_32.o
<instantiation>:7:9: error: invalid reassignment of non-absolute variable 'pos'
    pos=pos+8
        ^
./arch/x86/include/asm/idtentry.h:474:5: note: while in macro instantiation
    .rept (0xec - 0x20)
    ^
```
looks like there's a potentially large series that touched some of the related files from tglx on May 21.  Not sure yet if that's the culprit.

**Comment by nickdesaulniers** [comment]
unreproducible with
```sh
$ make LLVM=1 -j71
```
requires
```sh
$ make LLVM=1 LLVM_IAS=1 -j71
```
cc @jcai19 . We may have to look into disabling LLVM_IAS for x86_64 for linux-next.

**Comment by jcai19** [comment]
> looks like there's a potentially large series that touched some of the related files from tglx on May 21. Not sure yet if that's the culprit.

That should be the case. The complete code in arch/x86/include/asm/idtentry.h is as follows,

        .align 8
SYM_CODE_START(irq_entries_start)
    vector=FIRST_EXTERNAL_VECTOR
    pos = .
    .rept (FIRST_SYSTEM_VECTOR - FIRST_EXTERNAL_VECTOR)
        UNWIND_HINT_IRET_REGS
        .byte   0x6a, vector
        jmp     asm_common_interrupt
        nop
        /* Ensure that the above is 8 bytes max */
        . = pos + 8
    pos=pos+8
    vector=vector+1
    .endr
SYM_CODE_END(irq_entries_start)

Clang's integrated assembler imposes a restriction on symbols so that they are not allowed to be reassigned if their values are not absolute (such as constant). To fix this issue, we would either fix the code to not reassign pos and vector, or try to add support of reassigning symbols with non-absolute values.  I can't seem to find a workaround for the above code. But adding such support in IAS is not  a trivial task either. Currently IAS assumes each symbol is only assigned with non-absolute value at most once, and creates a symbol table while parsing assembly statements to record the value of each symbol. The values are then resolved and used to replace all the uses of the symbols after parsing is done, while IAS is finalizing the layout. It appears to me that to add such support would require IAS to record the uses of each definition of symbols, similar to constructing the SSA form. I wonder how GNU AS implements this.

**Comment by nickdesaulniers** [comment]
It sounds almost like these need to be in SSA form.  I wonder if temporary variables help here?  I'm also curious if `pos` and `vector` have uses outside of the final snippet you posted, otherwise the final assignment looks dead.

**Comment by jcai19** [comment]
Right, I think they were introduced just for this loop. Played with IAS and turned out it supported the following code

$ cat foo.s 
i=0
.rept 2
i=i+1
mov i, %eax
.endr

$ llvm-mc -triple=x86_64 -filetype=obj foo.s -o foo.o
$ objdump -d foo.o

foo.o:     file format elf64-x86-64


Disassembly of section .text:

0000000000000000 <.text>:
   0:	8b 04 25 01 00 00 00 	mov    0x1,%eax
   7:	8b 04 25 02 00 00 00 	mov    0x2,%eax

We should be able to rewrite the code using a loop counter to avoid using dot symbol.

**Comment by nickdesaulniers** [comment]
Please hold on sending. The second use of `FIRST_EXTERNAL_VECTOR` looks like maybe it should instead be `FIRST_SYSTEM_VECTOR`.

> Clang's integrated assembler imposes a restriction on symbols so that they are not allowed to be reassigned if their values are not absolute (such as constant).

It seems we can't then modify symbolic variables that were defined in terms of `.`. Hmm...

**Comment by jcai19** [comment]
> Please hold on sending. The second use of `FIRST_EXTERNAL_VECTOR` looks like maybe it should instead be `FIRST_SYSTEM_VECTOR`.

Yes. Thanks for the correction.

> It seems we can't then modify symbolic variables that were defined in terms of `.`. Hmm...

Right. The value of dot symbol is unknown to IAS while it's parsing the assembly statements.

**Comment by dileks** [comment]
Maybe I am caught by this in #1050 (see [1]).

In Linux v5.7.2 I do not see a `arch/x86/include/asm/idtentry.h` file.

[1] https://github.com/ClangBuiltLinux/linux/issues/1050#issuecomment-643200604

**Comment by nathanchance** [comment]
This is only a problem on -next, not stable or mainline.

I am running into this with my WSL2 kernel, I can test a patch before it is sent upstream.

**Comment by dileks** [comment]
As workaround with `Linux v5.8-rc1`:

```diff
diff --git a/arch/x86/entry/Makefile b/arch/x86/entry/Makefile
--- a/arch/x86/entry/Makefile
+++ b/arch/x86/entry/Makefile
@@ -14,6 +14,9 @@ CFLAGS_REMOVE_syscall_64.o = $(CC_FLAGS_FTRACE) -fstack-protector -fstack-protec
 CFLAGS_syscall_64.o            += $(call cc-option,-Wno-override-init,)
 CFLAGS_syscall_32.o            += $(call cc-option,-Wno-override-init,)
 obj-y                          := entry_$(BITS).o thunk_$(BITS).o syscall_$(BITS).o
+ifdef LLVM_IAS
+AFLAGS_entry_$(BITS).o         += -no-integrated-as
+endif
 obj-y                          += common.o
 
 obj-y                          += vdso/
```

Corresponding .o file is generated:
```
$ LC_ALL=C ll arch/x86/entry/entry_64.o
-rw-r--r-- 1 dileks dileks 76K Jun 16 20:42 arch/x86/entry/entry_64.o
```

**Comment by dileks** [comment]
@jcai19 

Any progress on this?

I am building with my workaround (see above) and `Linux v5.8-rc4` with `LLVM_IAS=1`.

**Comment by dileks** [comment]
Jupp, that worked.

```
$ llvm-dwarfdump-11 arch/x86/entry/entry_64.o
arch/x86/entry/entry_64.o:      file format elf64-x86-64

.debug_info contents:
0x00000000: Compile Unit: length = 0x0000001e, format = DWARF32, version = 0x0002, abbr_offset = 0x0000, addr_size = 0x08 (next unit at 0x00000022)

0x0000000b: DW_TAG_compile_unit
              DW_AT_stmt_list   (0x00000000)
              DW_AT_ranges      (0x00000000
                 [0x0000000000000000, 0x00000000000014ff)
                 [0x0000000000000000, 0x00000000000000b7)
                 [0x0000000000000000, 0x00000000000000d5)
                 [0x0000000000000000, 0x0000000000000013))
              DW_AT_name        ("arch/x86/entry/entry_64.S")
              DW_AT_comp_dir    ("/home/dileks/src/linux-kernel/git")
              DW_AT_producer    ("GNU AS 2.34")
              DW_AT_language    (DW_LANG_Mips_Assembler)
```

**Comment by jcai19** [comment]
> Jupp, that worked.

@dileks I assume "that" referred to your [workaround](https://github.com/ClangBuiltLinux/linux/issues/1043#issuecomment-644946765). Would you mind trying https://github.com/ClangBuiltLinux/linux/issues/1043#issuecomment-642163507? I think @nickdesaulniers had [concerns](https://github.com/ClangBuiltLinux/linux/issues/1043#issuecomment-642332869) regarding the patch. I will send it if it works and @nickdesaulniers is happy with it.

**Comment by dileks** [comment]
@jcai19 

Applied your diff.
I have stolen the original clang-11 IAS line and tested with/without `-no-integrated-as`:
```
 $ LC_ALL=C rm -v arch/x86/entry/entry_64.o
removed 'arch/x86/entry/entry_64.o'

$ clang-11 -Wp,-MMD,arch/x86/entry/.entry_64.o.d -nostdinc -isystem /usr/lib/llvm-11/lib/clang/11.0.0/include -I./arch/x86/include -I./arch/x86/include/generated  -I./include -I./arch/x86/include/uapi -I./arch/x86/include/generated/uapi -I./include/uapi -I./include/generated/uapi -include ./include/linux/kconfig.h -D__KERNEL__ -Qunused-arguments -D__ASSEMBLY__ -fno-PIE -Werror=unknown-warning-option -m64 -Wa,-gdwarf-4 -gz=zlib -DCC_USING_FENTRY -no-integrated-as    -c -o arch/x86/entry/entry_64.o arch/x86/entry/entry_64.S

$ LC_ALL=C ll arch/x86/entry/entry_64.o
-rw-r--r-- 1 dileks dileks 39K Jul  7 03:31 arch/x86/entry/entry_64.o

$ LC_ALL=C rm -v arch/x86/entry/entry_64.o
removed 'arch/x86/entry/entry_64.o'

$ clang-11 -Wp,-MMD,arch/x86/entry/.entry_64.o.d -nostdinc -isystem /usr/lib/llvm-11/lib/clang/11.0.0/include -I./arch/x86/include -I./arch/x86/include/generated  -I./include -I./arch/x86/include/uapi -I./arch/x86/include/generated/uapi -I./include/uapi -I./include/generated/uapi -include ./include/linux/kconfig.h -D__KERNEL__ -Qunused-arguments -D__ASSEMBLY__ -fno-PIE -Werror=unknown-warning-option -m64 -Wa,-gdwarf-4 -gz=zlib -DCC_USING_FENTRY     -c -o arch/x86/entry/entry_64.o arch/x86/entry/entry_64.S
<instantiation>:9:6: error: expected absolute expression
 .if HYPERVISOR_REENLIGHTENMENT_VECTOR == 3
     ^
<instantiation>:1:1: note: while in macro instantiation
idtentry HYPERVISOR_REENLIGHTENMENT_VECTOR asm_sysvec_hyperv_reenlightenment sysvec_hyperv_reenlightenment has_error_code=0
^
./arch/x86/include/asm/idtentry.h:627:1: note: while in macro instantiation
idtentry_sysvec HYPERVISOR_REENLIGHTENMENT_VECTOR sysvec_hyperv_reenlightenment;
^
<instantiation>:9:6: error: expected absolute expression
 .if HYPERVISOR_STIMER0_VECTOR == 3
     ^
<instantiation>:1:1: note: while in macro instantiation
idtentry HYPERVISOR_STIMER0_VECTOR asm_sysvec_hyperv_stimer0 sysvec_hyperv_stimer0 has_error_code=0
^
./arch/x86/include/asm/idtentry.h:628:1: note: while in macro instantiation
idtentry_sysvec HYPERVISOR_STIMER0_VECTOR sysvec_hyperv_stimer0;
^
```

**Comment by dileks** [comment]
I have `CONFIG_HYPERV=m`.

**Comment by dileks** [comment]
With `# CONFIG_HYPERV is not set`:

```
$ scripts/config -d HYPERV

  clang-11 -Wp,-MMD,arch/x86/entry/.entry_64.o.d -nostdinc -isystem /usr/lib/llvm-11/lib/clang/11.0.0/include -I./arch/x86/include -I./arch/x86/include/generated  -I./include -I./arch/x86/include/uapi -I./arch/x86/include/generated/uapi -I./include/uapi -I./include/generated/uapi -include ./include/linux/kconfig.h -D__KERNEL__ -Qunused-arguments -D__ASSEMBLY__ -fno-PIE -Werror=unknown-warning-option -m64 -Wa,-gdwarf-4 -gz=zlib -DCC_USING_FENTRY    -c -o arch/x86/entry/entry_64.o arch/x86/entry/entry_64.S
   ./tools/objtool/objtool orc generate  --no-fp --retpoline --uaccess arch/x86/entry/entry_64.o
  if llvm-objdump-11 -h arch/x86/entry/entry_64.o | grep -q __ksymtab; then { echo "#include <linux/kernel.h>" ; echo "#include <asm/asm-prototypes.h>" ; clang-11 -E -Wp,-MMD,arch/x86/entry/.entry_64.o.d -nostdinc -isystem /usr/lib/llvm-11/lib/clang/11.0.0/include -I./arch/x86/include -I./arch/x86/include/generated  -I./include -I./arch/x86/include/uapi -I./arch/x86/include/generated/uapi -I./include/uapi -I./include/generated/uapi -include ./include/linux/kconfig.h -D__KERNEL__ -Qunused-arguments -D__ASSEMBLY__ -fno-PIE -Werror=unknown-warning-option -m64 -Wa,-gdwarf-4 -gz=zlib -DCC_USING_FENTRY    arch/x86/entry/entry_64.S | grep "\<___EXPORT_SYMBOL\>" | sed 's/.*___EXPORT_SYMBOL[[:space:]]*\([a-zA-Z0-9_]*\)[[:space:]]*,.*/EXPORT_SYMBOL();/' ; } | clang-11 -E -D__GENKSYMS__ -Wp,-MMD,arch/x86/entry/.entry_64.o.d -nostdinc -isystem /usr/lib/llvm-11/lib/clang/11.0.0/include -I./arch/x86/include -I./arch/x86/include/generated  -I./include -I./arch/x86/include/uapi -I./arch/x86/include/generated/uapi -I./include/uapi -I./include/generated/uapi -include ./include/linux/kconfig.h -include ./include/linux/compiler_types.h -D__KERNEL__ -Qunused-arguments -Wall -Wundef -Werror=strict-prototypes -Wno-trigraphs -fno-strict-aliasing -fno-common -fshort-wchar -fno-PIE -Werror=implicit-function-declaration -Werror=implicit-int -Wno-format-security -std=gnu89 -Werror=unknown-warning-option -mno-sse -mno-mmx -mno-sse2 -mno-3dnow -mno-avx -m64 -mno-80387 -mstack-alignment=8 -mtune=generic -mno-red-zone -mcmodel=kernel -Wno-sign-compare -fno-asynchronous-unwind-tables -mretpoline-external-thunk -fno-delete-null-pointer-checks -Wno-frame-address -Wno-address-of-packed-member -O2 -Wframe-larger-than=2048 -fstack-protector-strong -Wno-format-invalid-specifier -Wno-gnu -mno-global-merge -Wno-unused-const-variable -g -gdwarf-4 -gz=zlib -pg -mfentry -DCC_USING_FENTRY -Wdeclaration-after-statement -Wvla -Wno-pointer-sign -Wno-array-bounds -fno-strict-overflow -fno-merge-all-constants -fno-stack-check -Werror=date-time -Werror=incompatible-pointer-types -fmacro-prefix-map=./= -fcf-protection=none -Wno-initializer-overrides -Wno-format -Wno-sign-compare -Wno-format-zero-length -Wno-pointer-to-enum-cast -Wno-tautological-constant-out-of-range-compare    -DKBUILD_MODFILE='"arch/x86/entry/entry_64"' -DKBUILD_BASENAME='"entry_64"' -DKBUILD_MODNAME='"entry_64"' -xc - | scripts/genksyms/genksyms    -r /dev/null > arch/x86/entry/.tmp_entry_64.ver; ld.lld-11 -m elf_x86_64  -z max-page-size=0x200000 --compress-debug-sections=zlib -r -o arch/x86/entry/.tmp_entry_64.o arch/x86/entry/entry_64.o -T arch/x86/entry/.tmp_entry_64.ver; mv -f arch/x86/entry/.tmp_entry_64.o arch/x86/entry/entry_64.o; rm -f arch/x86/entry/.tmp_entry_64.ver; fi

$ LC_ALL=C ll arch/x86/entry/entry_64.o
-rw-r--r-- 1 dileks dileks 83K Jul  7 04:13 arch/x86/entry/entry_64.o
```

**Comment by dileks** [comment]
Switched over to Debian's `binutils version 2.34.90.20200706-1` which is able to produce `DWARF-4` (see: `version = 0x0004`):

```
$ llvm-dwarfdump-11 arch/x86/entry/entry_64.o | head -20
arch/x86/entry/entry_64.o:      file format elf64-x86-64

.debug_info contents:
0x00000000: Compile Unit: length = 0x0000001e, format = DWARF32, version = 0x0004, abbr_offset = 0x0000, addr_size = 0x08 (next unit at 0x00000022)

0x0000000b: DW_TAG_compile_unit
              DW_AT_stmt_list   (0x00000000)
              DW_AT_ranges      (0x00000000
                 [0x0000000000000000, 0x00000000000014ff)
                 [0x0000000000000000, 0x00000000000000b7)
                 [0x0000000000000000, 0x00000000000000d5)
                 [0x0000000000000000, 0x0000000000000013))
              DW_AT_name        ("arch/x86/entry/entry_64.S")
              DW_AT_comp_dir    ("/home/dileks/src/linux-kernel/git")
              DW_AT_producer    ("GNU AS 2.34.90")
              DW_AT_language    (DW_LANG_Mips_Assembler)
```

**Comment by jcai19** [comment]
> <instantiation>:9:6: error: expected absolute expression
>  .if HYPERVISOR_REENLIGHTENMENT_VECTOR == 3
>      ^
> <instantiation>:1:1: note: while in macro instantiation
> idtentry HYPERVISOR_REENLIGHTENMENT_VECTOR asm_sysvec_hyperv_reenlightenment sysvec_hyperv_reenlightenment has_error_code=0
> ^

This looks like a different issue. Can you please file a separate bug to track that? Thanks. My patch only fixed the error as reported, i.e. invalid reassignment of non-absolute variable 'pos'. I just verified entry_64.S assembled with IAS on mainline,  with the patch applied. Interesting, assembling the same file on linux-next with the patch applied still failed with the following message. Will file a separate bug.


clang -Wp,-MMD,arch/x86/entry/.entry_64.o.d  -nostdinc -isystem ${PATH_TO_LLVM_SRC}/build/release/lib/clang/11.0.0/include -I./arch/x86/include -I./arch/x86/include/generated  -I./include -I./arch/x86/include/uapi -I./arch/x86/include/generated/uapi -I./include/uapi -I./include/generated/uapi -include ./include/linux/kconfig.h -D__KERNEL__ -Qunused-arguments -D__ASSEMBLY__ -fno-PIE -Werror=unknown-warning-option -m64    -c -o arch/x86/entry/entry_64.o arch/x86/entry/entry_64.S
<instantiation>:2:2: error: unknown use of instruction mnemonic without a size suffix
 lsl %rax, %rax
 ^
<instantiation>:2:2: note: while in macro instantiation
 LOAD_CPU_AND_NODE_SEG_LIMIT %rax
 ^
<instantiation>:1:1: note: while in macro instantiation
ALTERNATIVE "LOAD_CPU_AND_NODE_SEG_LIMIT %rax", "RDPID  %rax", (16*32+22)
^
<instantiation>:2:2: note: while in macro instantiation
 GET_PERCPU_BASE %rax
 ^
arch/x86/entry/entry_64.S:846:2: note: while in macro instantiation
 SAVE_AND_SET_GSBASE scratch_reg=%rax save_reg=%rbx

**Comment by dileks** [comment]
@jcai19 

Thanks for double-checking.

I agree this a different issue - as demonstrated with `CONFIG_HYPERV={m,y}`.

I kept my workaround as this means to me:  `CONFIG_HYPERV=n`

**Comment by dileks** [comment]
@jcai19 

Can you tell me how you "verified ... on mainline"?
What do you mean by "mainline" - `Linux v5.8-rc4`?

Between Linux v5.8-rc3 and v5.8-rc4 there was a lot of `x86/entry` stuff added.
That's why I ask for.

Just curious: Which llvm-toolchain version do you use?

Here:
```
# clang-11 -v
Debian clang version 11.0.0-++20200701093119+ffee8040534-1~exp1 
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/10
Found candidate GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/9
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/10
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/9
Selected GCC installation: /usr/bin/../lib/gcc/x86_64-linux-gnu/10
Candidate multilib: .;@m64
Candidate multilib: 32;@m32
Candidate multilib: x32;@mx32
Selected multilib: .;@m64
```

**Comment by jcai19** [comment]
> Can you tell me how you "verified ... on mainline"?

So I ran the following commands, and without/with the patch, they failed/passed respectively.

```
$ ARCH=x86_64 make clean
$ ARCH=x86_64 make CC=clang LLVM_IAS=1 defconfig
$ ARCH=x86_64 make -j32 CC=clang LLVM_IAS=1 V=1 arch/x86/entry/entry_64.o
```

> What do you mean by "mainline" - `Linux v5.8-rc4`?

Yes, I was referring to https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/?h=v5.8-rc4. My local copy was on dcde237b9b0eb1d19306e6f48c0a4e058907619f.

> Just curious: Which llvm-toolchain version do you use?


I was using LLVM TOT. 

${PATH_TO_LLVM_SRC}build/release/bin$ ./clang -v
clang version 11.0.0 (https://github.com/llvm/llvm-project.git b925ca37a8f28851e110b8f0cd6e7f9a36a15d65)
Target: x86_64-unknown-linux-gnu
Thread model: posix
InstalledDir: /usr/local/google/home/jiancai/storage/llvm-project/build/release/bin/.
Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/10
Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/8
Found candidate GCC installation: /usr/lib/gcc/i686-linux-gnu/9
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/10
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/6.5.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/7
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/7.5.0
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/8
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/9
Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/9
Candidate multilib: .;@m64
Selected multilib: .;@m64

**Comment by dileks** [comment]
@jcai19 

I kept my last working build-environment and Linux-kernel setup.

Applied your patch but kept my `no-integrated-as` workaround - build and boot on bare metal is OK.

Now, I do a 2nd run with my workaround reverted plus `CONFIG_HYPERV=n` - doing a new full build.

Will report later if I am able to boot on bare metal.

**Comment by dileks** [comment]
> > :9:6: error: expected absolute expression
> > .if HYPERVISOR_REENLIGHTENMENT_VECTOR == 3
> > ^
> > :1:1: note: while in macro instantiation
> > idtentry HYPERVISOR_REENLIGHTENMENT_VECTOR asm_sysvec_hyperv_reenlightenment sysvec_hyperv_reenlightenment has_error_code=0
> > ^
> 
> This looks like a different issue. Can you please file a separate bug to track that? Thanks. My patch only fixed the error as reported, i.e. invalid reassignment of non-absolute variable 'pos'. I just verified entry_64.S assembled with IAS on mainline, with the patch applied. 

As desired I opened a new isssue #1088.

**Comment by dileks** [comment]
@jcai19 

Typos :-) see https://github.com/ClangBuiltLinux/linux/issues/1088#issuecomment-657250733

**Comment by jcai19** [comment]
@dileks Thanks for reporting and looking into the issue!

**Comment by dileks** [comment]
@jcai19 

What are your plans for your proposed fix here?
Fix this in `Linux v5.8-rc4+` and/or `Linux-next`?

**Comment by jcai19** [comment]
> Fix this in `Linux v5.8-rc4+` and/or `Linux-next`?

I just sent the patch for review. I don't know which one it will go, probably Linux-next based on my limited experience?

**Comment by dileks** [comment]
Patch submitted.

[1] https://lore.kernel.org/patchwork/patch/1271455/
[2] https://lore.kernel.org/lkml/20200713012428.1039487-1-caij2003@gmail.com/

**Comment by jcai19** [comment]
@dileks FYI I just updated the patch based on comments and kept your tested-by tag. Feel free to test it again :)

**Comment by dileks** [comment]
@jcai19 submitted a v2.

[1] https://lore.kernel.org/patchwork/patch/1273067/

**Comment by nickdesaulniers** [comment]
accepted: https://git.kernel.org/pub/scm/linux/kernel/git/tip/tip.git/commit/?id=6ee93f8df09c470da1a4af11e394c52d7b62418c

**Comment by dileks** [comment]
Congrats @jcai19

**Comment by jcai19** [comment]
@dileks Thanks for your and @nickdesaulniers's help. And congrats on your [patch](https://git.kernel.org/pub/scm/linux/kernel/git/tip/tip.git/commit/?id=5769fe26f389b0002ed48fd16d642a1d86edaf79) landing on kernel :)

**Comment by dileks** [comment]
Accepted in Linus tree.

https://git.kernel.org/linus/6ee93f8df09c470da1a4af11e394c52d7b62418c

**Comment by jcai19** [comment]
Thanks for the update!

**Comment by MaskRay** [comment]
From https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/commit/?id=d72c4a36d7ab560127885473a310ece28988b604

> There is no hope of getting this fixed in LLVM (see https://github.com/ClangBuiltLinux/linux/issues/1043#issuecomment-641571200 and https://bugs.llvm.org/show_bug.cgi?id=47798#c1 ) so if we want to build with LLVM_IAS, we need to hack
around it ourselves.

Fixed for LLVM 21.1:) https://github.com/llvm/llvm-project/commit/e015626f189dc76f8df9fdc25a47638c6a2f3feb
